brian2/devices/cpp_standalone/device.py

```diff
@@ -1400,20 +1401,58 @@ def network_run(self, net, duration, report=None, report_period=10*second,
                 for codeobj in obj._code_objects:
                     code_objects.append((obj.clock, codeobj))
 
+        #Code for time formatting function  
+        time_format_func = '''
+        std::string format_time(int time_in_s)
+        {
+            int hours = time_in_s / 3600;
+
+            time_in_s = time_in_s % 3600;
+            int minutes = time_in_s / 60;
+
+            time_in_s = time_in_s % 60;
+
+            std::stringstream ss,ms,hs;
+            hs << hours;
+            std::string hrs = hs.str();
+
+            ms << minutes;
+            std::string mins = ms.str();
+            
+            ss << time_in_s;
+            std::string secs = ss.str();
+
+            if(hours < 1){
+                if(minutes < 1){
+                    return secs + "s";
+                }else{
+                    return mins + "m " + secs + "s"; 
+                }
+            }else{
+                return hrs + "h " + mins + "m " + secs + "s";
+            }
+        }'''
+
         # Code for a progress reporting function
         standard_code = '''
         void report_progress(const double elapsed, const double completed, const double start, const double duration)
         {
+            std::string duration_format = format_time((int)duration);
```

Comment:
`duration_format` is not used anywhere, so it can be deleted.

Reply:
I've used duration_format at line 1446 to format the duration.

Reply:
Should the duration be formatted?

Reply:
No, `duration` (and `completed*duration`) refer to "biological time" and are in seconds or even milliseconds and should not be formatted this way. Only the  `elapsed` and the `remaining` time are "wall clock time" and are seconds/minutes/hours/...

Reply:
Okay.